pkg/fortify/fortify.go

```diff
@@ -0,0 +1,686 @@
+)
+
+// System is the interface abstraction of a specific SystemInstance
+type System interface {
+	GetProjectByName(name string, autoCreate bool, projectVersion string) (*models.Project, error)
+	GetProjectVersionDetailsByProjectIDAndVersionName(id int64, name string, autoCreate bool, projectName string) (*models.ProjectVersion, error)
+	GetProjectVersionAttributesByProjectVersionID(id int64) ([]*models.Attribute, error)
+	SetProjectVersionAttributesByProjectVersionID(id int64, attributes []*models.Attribute) ([]*models.Attribute, error)
+	CreateProjectVersionIfNotExist(projectName, projectVersionName, description string) (*models.ProjectVersion, error)
+	LookupOrCreateProjectVersionDetailsForPullRequest(projectID int64, masterProjectVersion *models.ProjectVersion, pullRequestName string) (*models.ProjectVersion, error)
+	CreateProjectVersion(version *models.ProjectVersion) (*models.ProjectVersion, error)
+	ProjectVersionCopyFromPartial(sourceID, targetID int64) error
+	ProjectVersionCopyCurrentState(sourceID, targetID int64) error
+	ProjectVersionCopyPermissions(sourceID, targetID int64) error
+	CommitProjectVersion(id int64) (*models.ProjectVersion, error)
+	MergeProjectVersionStateOfPRIntoMaster(downloadEndpoint, uploadEndpoint string, masterProjectID, masterProjectVersionID int64, pullRequestName string) error
+	GetArtifactsOfProjectVersion(id int64) ([]*models.Artifact, error)
+	GetFilterSetOfProjectVersionByTitle(id int64, title string) (*models.FilterSet, error)
+	GetIssueFilterSelectorOfProjectVersionByName(id int64, names []string, options []string) (*models.IssueFilterSelectorSet, error)
+	GetProjectIssuesByIDAndFilterSetGroupedBySelector(id int64, filter, filterSetGUID string, issueFilterSelectorSet *models.IssueFilterSelectorSet) ([]*models.ProjectVersionIssueGroup, error)
+	ReduceIssueFilterSelectorSet(issueFilterSelectorSet *models.IssueFilterSelectorSet, names []string, options []string) *models.IssueFilterSelectorSet
+	GetIssueStatisticsOfProjectVersion(id int64) ([]*models.IssueStatistics, error)
+	GenerateQGateReport(projectID, projectVersionID, reportTemplateID int64, projectName, projectVersionName, reportFormat string) (*models.SavedReport, error)
+	GetReportDetails(id int64) (*models.SavedReport, error)
+	UploadResultFile(endpoint, file string, projectVersionID int64) error
+	DownloadReportFile(endpoint string, projectVersionID int64) ([]byte, error)
+	DownloadResultFile(endpoint string, projectVersionID int64) ([]byte, error)
+}
+
+// SystemInstance is the specific instance
+type SystemInstance struct {
+	timeout    time.Duration
+	token      string
+	serverURL  string
+	client     *ff.Fortify
+	httpClient *piperHttp.Client
+	logger     *logrus.Entry
+}
+
+// NewSystemInstance - creates an returns a new SystemInstance
+func NewSystemInstance(serverURL, endpoint, authToken string, timeout time.Duration) *SystemInstance {
+	schemeHost := strings.Split(serverURL, "://")
+	hostEndpoint := strings.Split(schemeHost[1], "/")
```

Comment:
These parameters are passed directly from [here](https://github.com/SAP/jenkins-library/pull/1428/files#diff-875f581bf1d0c73d908d7f634cd2298eR37) and are not checked at all. If they are malformed, piper would just crash, I assume.